### PR TITLE
Bugfix/fix(LIVE-10851)

### PR DESCRIPTION
**Pull request by CremaFR**
### 📝 Description

https://github.com/LedgerHQ/ledger-live/assets/5360522/bd99cc71-7751-4a4b-afc5-e4984af51d41

### ❓ Context

[- **JIRA link**: ](https://ledgerhq.atlassian.net/browse/LIVE-10851)

### ✅ Checklist

Pull Requests must pass the CI and be code reviewed. Set as Draft if the PR is not ready.

- [x] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - 

---

### 🧐 Checklist for the PR Reviewers

- [ ] **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- [ ] **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- [ ] **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- [ ] **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- [ ] **Any new dependencies** have been justified and documented.
- [ ] **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

File: apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/index.tsx [hunk 1]
```diff
@@ -118,15 +118,24 @@ const SwapForm = () => {
   }, []);
 
   const exchangeRatesState = swapTransaction.swap?.rates;
-  const swapError = swapTransaction.fromAmountError || exchangeRatesState?.error;
+  let swapError = swapTransaction.fromAmountError || exchangeRatesState?.error;
```
Comment (cpellizzari-ledger) [inline apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/index.tsx:121]: for readability of the code : 
- having "const" is better, I know it's definitive.
- I like to read small functions

I suggest you let const here, ending with || detectSpecificCoinError()
detectSpecificCoinError() may switch on the family first

**Review by cpellizzari-ledger — APPROVED** [review]

File: apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/index.tsx (hunk 1 above)
Comment (CremaFR) [reply apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/index.tsx:121]: Yeah you're right, extracted tezos check in its own function

**Review by cpellizzari-ledger — APPROVED** [review]

**Review by kallen-ledger — APPROVED** [review]

File: .changeset/olive-llamas-collect.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"ledger-live-desktop": patch
```
Comment (cpellizzari-ledger) [inline .changeset/olive-llamas-collect.md:2]: it's also LLM patch

File: apps/ledger-live-mobile/src/screens/Swap/Form/index.tsx [hunk 3]
```diff
@@ -121,7 +125,22 @@ export function SwapForm({
     );
   }, [exchangeRatesState.value, swapTransaction.swap.to.currency]);
 
-  const swapError = swapTransaction.fromAmountError || exchangeRatesState?.error;
+  const isTezosAccountReveal = (swapTransaction: SwapTransactionType): Error | undefined => {
```
Comment (cpellizzari-ledger) [inline apps/ledger-live-mobile/src/screens/Swap/Form/index.tsx:128]: could you move this fn in LL common ?
as for name, I would rather have "maybeTezosAccountUnrevealedAccount". This name looks more like a test (with bool result)

File: .changeset/olive-llamas-collect.md (hunk 2 above)
Comment (CremaFR) [reply .changeset/olive-llamas-collect.md:2]: done and added also common with the refactor :)

**Review by kallen-ledger — APPROVED** [review]

**Review by kallen-ledger — APPROVED** [review]

**Pull request merged**